packages/SwingSet/src/kernel/kernel.js

```diff
@@ -303,11 +303,10 @@ export default function buildKernel(kernelEndowments) {
   async function deliverToVat(vatID, target, msg) {
     insistMessage(msg);
     const vat = ephemeral.vats.get(vatID);
-    assert(vat, details`unknown vatID ${vatID}`);
     kernelKeeper.incStat('dispatches');
     kernelKeeper.incStat('dispatchDeliver');
-    if (vat.dead) {
-      resolveToError(msg.result, makeError('vat is dead'));
+    if (!vat || vat.dead) {
```

Comment:
this is fine for now, in the long run it'd be nice to have exactly one way to specify that a vat is dead, instead of two

Reply:
I think `vat is dead` is the better error message.. userspace doesn't get to pick the vatid, so telling them that their choice was wrong seems to be sending the error at the wrong level. They only way they should get this error is if the object they sent to is in a vat which died.

Reply:
I think the `vat.dead` branch of the OR will go away if we follow your suggestion of simply removing the `ephemeral.vats` entry and instead provide a kernel keeper method to check directly if it has a vat or not.